pkg/k8s/annotate.go

```diff
@@ -95,40 +96,32 @@ func (k8sCli K8sClient) AnnotateNode(nodeName string, v4CIDR, v6CIDR *cidr.CIDR,
 	})
 	scopedLog.Debug("Updating node annotations with node CIDRs")
 
-	go func(c kubernetes.Interface, nodeName string, v4CIDR, v6CIDR *cidr.CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP net.IP) {
-		var node *v1.Node
-		var err error
-
-		for n := 1; n <= maxUpdateRetries; n++ {
-			if node == nil {
-				node, err = GetNode(c, nodeName)
+	controller.NewManager().UpdateController("update-k8s-node-annotations",
+		controller.ControllerParams{
+			DoFunc: func(_ context.Context) error {
+				node, err := GetNode(k8sCli, nodeName)
 				if errors.IsNotFound(err) {
 					err = ErrNilNode
 				}
-			}
-
-			if err == nil && node != nil {
-				node, err = updateNodeAnnotation(c, node, v4CIDR, v6CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP)
-			}
-
-			switch {
-			case err == nil:
-				return
-			case errors.IsConflict(err):
-				scopedLog.WithFields(logrus.Fields{
-					fieldRetry:    n,
-					fieldMaxRetry: maxUpdateRetries,
-				}).WithError(err).Debugf("Unable to update node resource with annotation")
-			default:
-				scopedLog.WithFields(logrus.Fields{
-					fieldRetry:    n,
-					fieldMaxRetry: maxUpdateRetries,
-				}).WithError(err).Warn("Unable to update node resource with annotation")
-			}
-
-			time.Sleep(time.Duration(n) * time.Second)
-		}
-	}(k8sCli, nodeName, v4CIDR, v6CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP)
+
+				if err == nil && node != nil {
+					_, err = updateNodeAnnotation(k8sCli, node, v4CIDR, v6CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP)
+				}
+
+				switch {
+				case err == nil:
+					return nil
+				case errors.IsConflict(err):
+					scopedLog.WithFields(logrus.Fields{
+						fieldMaxRetry: maxUpdateRetries,
+					}).WithError(err).Debugf("Unable to update node resource with annotation")
+					return err
```

Comment:
This is not good, it will lead to a lot of controllers to be reported as failing if the Update conflicts. Conflicting updates is not an error condition unless it happens several times in a row. Please change it to do a couple of retries before failing.

Reply:
@tgraf but the retry should be made by the controller. Don't care to much about this as there's a PR coming that deprecates all of this code.

Reply:
OK, if we are going deprecate, then I'm fine.